src/devices/Ssd13xx/Ssd1309.cs

```diff
@@ -0,0 +1,226 @@
+            }
+        }
+
+        /// <summary>
+        /// Sends data to the device
+        /// </summary>
+        /// <param name="data">Data being sent</param>
+        public override void SendData(Span<byte> data)
+        {
+            if (data.IsEmpty)
+            {
+                throw new ArgumentNullException(nameof(data));
+            }
+
+            if (SpiDevice != null)
+            {
+                // Begin consuming SPI bus data
+                // Because the timing is not perfect, this may have adverse side-effects when using multiple SPI bus devices
+                _csGpioPin.Write(PinValue.Low);
+
+                // Enable data mode
+                _dcGpioPin.Write(PinValue.High);
+                SpiDevice.Write(data);
+
+                // Stop consuming SPI bus data
+                _csGpioPin.Write(PinValue.High);
+            }
+            else
+            {
+                throw new InvalidOperationException("No SPI device available or it has been disposed.");
+            }
+        }
+
+        /// <summary>
+        /// Init 128x64
+        /// </summary>
+        protected virtual void Initialize()
+        {
+            SendCommand(new SetDisplayOff());
+            SendCommand(new SetDisplayClockDivideRatioOscillatorFrequency());
+            SendCommand(new SetMultiplexRatio());
+            SendCommand(new SetDisplayOffset());
+            SendCommand(new SetDisplayStartLine());
+            SendCommand(new SetMemoryAddressingMode(SetMemoryAddressingMode.AddressingMode.Horizontal));
+            SetStartAddress();
+            SendCommand(new SetSegmentReMap(true));
+            SendCommand(new SetComOutputScanDirection(false));
+            SendCommand(new SetComPinsHardwareConfiguration());
+            SendCommand(new SetContrastControlForBank0());
+            SendCommand(new SetPreChargePeriod());
+            SendCommand(new SetVcomhDeselectLevel());
+            SendCommand(new EntireDisplayOn(false));
+            SendCommand(new SetNormalDisplay());
+            SendCommand(new SetDisplayOn());
+            Thread.Sleep(200);
+            ClearScreen();
+        }
+
+        /// <summary>
+        /// Set the start address for the display
+        /// </summary>
+        protected override void SetStartAddress()
+        {
+            SendCommand(new SetColumnAddress());
+
+            if (ScreenHeight == 32)
+            {
+                SendCommand(new SetPageAddress(PageAddress.Page0, PageAddress.Page3));
+            }
+            else
+            {
+                SendCommand(new SetPageAddress());
+            }
+        }
+
+        /// <summary>
+        /// Reset device by cycling RST pin to LOW and resting at HIGH
+        /// </summary>
+        protected virtual void Reset()
+        {
+            _rstGpioPin.Write(PinValue.Low);
+            Thread.Sleep(100);
+            _rstGpioPin.Write(PinValue.High);
+            Thread.Sleep(100);
```

Comment:
This method was intended to just execute the specific sequence to reset the controller according to the datasheet. But practically, it could be useful to include other reset-related tasks. Calling `Reset` and then `Initialize` separately in a "re-initialization" procedure feels more intuitive to me personally, but I can see advantages to both.